src/arg-parser.ts

```diff
@@ -0,0 +1,103 @@
+import type { OptionDefinition } from 'command-line-usage';
+import * as commandLineArgs from 'command-line-args';
+
+function fail(msg: string): never {
+  console.error(msg);
+  process.exit(1);
+}
+
+export function parse<T extends readonly OptionDefinition[]>(
+  options: T,
+  printHelp: () => void
+): ArgsFromOptions<T> {
+  const def = options.find(o => o.defaultOption);
+  if (!def?.multiple) {
+    // this is just so I don't have to think about how to handle `--` in other cases
+    // not an inherent limitation
+    throw new Error('ecmarkup arg-parser requires a default option for now');
```

Comment:
This error doesn't help the end user resolve the issue. What are they supposed to do in response to this? Or will this not actually occur in practice?

Reply:
This function is only called internally, so it doesn't occur in practice. This is an error you'd only run into if as a developer of ecmarkup, changing the arguments to ecmarkup's CLI.